ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/output/PreventiveAndCurativesRaoOutput.java

```diff
@@ -207,14 +205,16 @@ public boolean isActivatedDuringState(State state, RangeAction rangeAction) {
     public int getPreOptimizationTapOnState(State state, PstRangeAction pstRangeAction) {
         if (state.getInstant() == Instant.PREVENTIVE) {
             return initialResult.getOptimizedTap(pstRangeAction);
-        } else {
+        } else if (postCurativeResults.containsKey(state)) {
             return postPreventiveResult.getOptimizedTap(pstRangeAction);
+        } else {
+            throw new FaraoException(String.format("State %s was not optimized and does not have pre-optim values", state.getId()));
         }
     }
 
     @Override
     public int getOptimizedTapOnState(State state, PstRangeAction pstRangeAction) {
-        if (state.getInstant() == Instant.PREVENTIVE) {
+        if (state.getInstant() == Instant.PREVENTIVE || !postCurativeResults.containsKey(state)) {
```

Comment:
why shouldn't we throw an error if !postCurativeResults.containsKey(state)
like what is done with methods getPreOptimizationTap/SetpointOnState

Reply:
this allows the user to access the optimal PST setpoint in a given a state that was not optimized in curative (by returning the optimal tap post-preventive).
I think it is more user-friendly than throwing an exception.

On the other hand, trying to getPreOptimTap/setpoint means the user supposes that the state was optimized in curative, so throwing an exception is less problematic